ledger/src/blockstore_meta.rs

```diff
@@ -125,9 +125,8 @@ pub struct ErasureMeta {
     set_index: u64,
     /// First coding index in the FEC set
     first_coding_index: u64,
-    /// Size of shards in this erasure set
-    #[serde(rename = "size")]
-    __unused_size: usize,
+    /// Index of the first received coding shred in the FEC set
+    first_received_coding_index: usize,
```

Comment:
Assuming this change first lands in v2.0 (exact branches don't matter, but just for sake of conversation):
- Node running with v1.18 inserts a bunch of these structs with unused field (value set to 0)
- Node running with v2.0 reads an `ErasureMeta` that had been inserted when node was running v1.18 - it reads back `first_received_coding_index` as 0

Could this domino into a problematic situation? Ie, we try to go fetch shred coding shred with `index=0` in the FEC set, but it isn't actually present ?

Reply:
the follow up PR that uses `first_received_coding_index()` https://github.com/anza-xyz/agave/pull/835/files should ensure that this is not possible.
We only call `first_received_coding_index()` on erasure meta's that were *just* created and are pending insertion into blockstore. This ensures that this field on erasure metas created with a previous version will not be used.

Reply:
this does bring up a good point for a future improvement I was planning - using `first_received_coding_index` to minimize the blockstore lookups here: https://github.com/anza-xyz/agave/blob/862c79e132620c6739e0b8db8b5f1972d0ee647c/ledger/src/blockstore.rs#L1420-L1421
will keep this in mind for the upcoming pr.